lorax/lorax.py

```diff
@@ -33,24 +34,340 @@ class TheLorax(object):
-        test_mat (pandas.DataFrame) The test matrix containing all examples to be
-            explained. If `id_col=None` (the default), the id for referencing entities
-            must be set as this dataframe's index.
-        id_col (str) The column name for the entity id in the test matrix. If `None`
-            (the default), the test matrix must be indexed by the entity id.
-        date_col (str) The date column in the matrix (default: `as_of_date`)
-        outcome_col (str) The outcome column in the matrix (default: `outcome`). To
-            indicate that the test matrix has no labels, set `outcome_col=None`.
-        name_patterns (list) An optional list of regex patterns or compiled regex
+            The classifier should be already trained. 
+        
+        column_names (List(str)): The input feature names of the data for which the classifier was trained
+
+        column_patterns (list): An optional list of regex patterns or compiled regex
             objects to group together features for reporting contributions. If using,
             each feature name in the test matrix must match one and only one pattern.
+        
+        test_mat (pandas.DataFrame): A test matrix to be pre-loaded to the object. This dataset will be used for providing feature distributions to the individual
+            If `id_col=None` (the default), the id for referencing entities
+            must be set as this dataframe's index.
+
+        id_col (str): The column name for the entity id in the test matrix. If `None`
+            (the default), the test matrix must be indexed by the entity id.
+        
+        date_col (str): The date column in the matrix (default: None).
+            If None, the matrix shouldn't contain a column with date
+        
+        outcome_col (str): The outcome column in the matrix (default: None). 
+            To indicate that the test matrix has no labels, set `outcome_col=None`.
+    
     """
 
-    def __init__(self, clf, test_mat, id_col=None,
+    def __init__(self, clf, column_names,
+                 column_patterns=None,
+                 test_mat=None,
+                 id_col=None,
+                 date_col=None, outcome_col=None):
+
+        self.clf = clf
+
+        # NOTE: Minor. maybe this should be feature_names and feature_patterns 
+        # To separate from the index and the outcome
+        self.column_names = column_names
+        self.column_patterns = column_patterns
+        
+        # Register the regex patterns and associated columns if using
+        if column_patterns is not None:
+            self.set_name_patterns(column_patterns)
+
+        # NOTE-KA: I feel like the method should be independent of these as these seem very triage specific. 
+        # We can always have a script that bridges the triage data with the explain API
+        # Leaving this decoupling to another PR
+        self.id_col = id_col
+        self.date_col = date_col
+        self.outcome_col = outcome_col
```

Comment:
That's fair, though probably most true for the date column. I can't remember if we use the date anywhere as such, otherwise, we could maybe simple change this to a list of columns (in addition to the id and outcome) to be excluded from analysis...